lib/time/DateTime.js

```diff
@@ -46,4 +46,9 @@ DateTime.fromJSON = function fromJSON(value) {
   return DateTime.fromSQL(value);
 };
 
+DateTime.fromLocaleString = function fromLocaleString(str) {
+  // TODO: get format from locale
```

Comment:
...actually, this turns out to be a limitation of `Intl.DateTimeFormat`: https://github.com/tc39/ecma402/issues/342

Since `luxon` is really just a thin wrapper around this API, it's not clear that this would be super-easy for them to address.